DetectorDescription/DDCMS/plugins/DDHGCalCell.cc

```diff
@@ -0,0 +1,288 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+
+static long  algorithm(dd4hep::Detector& /* description */,
+                       cms::DDParsingContext& ctxt,
+                       xml_h e,
+                       dd4hep::SensitiveDetector& /* sens */) {
+
+  cms::DDNamespace      ns(ctxt,e,true);
+  cms::DDAlgoArguments  args(ctxt, e);
+  // Header section of original DDHGCalCell.h
+  double                   waferSize  = args.value<double>("WaferSize");    // size of wafer
+  double                   waferT     = args.value<double>("WaferThick");   // Thickness of wafer
+  double                   cellT      = args.value<double>("CellThick");    // Thickness of depletion layer
+  int                      nCells     = args.value<int>("NCells");          // Number of cells 
+  int                      posSens    = args.value<int>("PosSensitive");    // Position of delpletion layer within wafer
+  std::string              material   = args.value<std::string>("Material");// Material
+  std::string              fullCN     = args.value<std::string>("FullCell");// Name of the full cell
+  std::string              fullSensN  = args.value<std::string>("FullSensitive");// Name of the sensitive part for a full cell
+  std::vector<std::string> truncCN    = args.value<std::vector<std::string> >("TruncatedCell");       // Names of truncated cells
+  std::vector<std::string> truncSensN = args.value<std::vector<std::string> >("TruncatedSensitive");  // Names of the sensitive parts for truncated cells
+  std::vector<std::string> extenCN    = args.value<std::vector<std::string> >("ExtendedCell");        // Names of extended cells
+  std::vector<std::string> extenSensN = args.value<std::vector<std::string> >("ExtendedSensitive");   // Names of the sensitive parts for extended cells
+  std::vector<std::string> cornrCN    = args.value<std::vector<std::string> >("CornerCell");          // Names of the corner cells
+  std::vector<std::string> cornrSensN = args.value<std::vector<std::string> >("CornerSensitive");     // Names of the sensitive parts for corner cells
+
+  if ((truncCN.size() != truncSensN.size()) ||
+      (extenCN.size() != extenSensN.size()) ||
+      (cornrCN.size() != cornrSensN.size()))
+    edm::LogWarning("HGCalGeom") << "The number of cells & sensitive differ:"
+				 << " Truncated " << truncCN.size() << ":"
+				 << truncSensN.size() << " Extended "
+				 << extenCN.size() <<":" << extenSensN.size()
+				 << " Corners " << cornrCN.size() << ":"
+				 << cornrSensN.size();
+  if ((truncCN.size() != 3) || (extenCN.size() != 3) ||
+      (cornrCN.size() != 6))
```

Comment:
Shouldn't there be a `return` statement for this `if`? If `truncCN.size()` is greater than 3, the code below will read way out of bounds. Currently, the code just issues a warning and marches along into memory errors.

Reply:
I am sending an exception